stl/inc/variant

```diff
@@ -1460,29 +1466,38 @@ struct _Variant_dispatcher;
 
 template <size_t... _Is>
 struct _Variant_dispatcher<index_sequence<_Is...>> {
-    template <class _Callable, class... _Types, bool _Any_valueless = ((_Is == 0) || ...)>
-    _NODISCARD static constexpr _Variant_visit_result_t<_Callable, _Types...> _Dispatch(
-        _Callable&& _Obj, _Types&&... _Args) {
+    template <class _Ret, class _Callable, class... _Types, bool _Any_valueless = ((_Is == 0) || ...)>
+    _NODISCARD static constexpr _Ret _Dispatch(_Callable&& _Obj, _Types&&... _Args) {

```

Comment:
Our name-mangling doesn't track which types came from parameter packs - unlike the Itanium ABI (https://godbolt.org/z/1ubA_c) - so this could potentially link to an old/wrong specialization of `_Dispatch` in another TU. I can't imagine such a contrived situation happening in the real world, but we should change the name to `_Dispatch2` here nonetheless to avoid this miniscule ABI break.

Reply:
Should `_Dispatch2`  be wrapped in a `#if _CXX20` block while keeping `_Dispatch` with C++17?